mesheryctl/internal/cli/root/system/channel.go

```diff
@@ -81,11 +82,62 @@ var viewCmd = &cobra.Command{
 	},
 }
 
+//check release channel version string supplied by user for validity. This func does not set any thing.(Just for checks)
+func validVersionCheck(c string) (bool, error) {
+	stableRegx := regexp.MustCompile(`^stable-v[0-9]+\.[0-9]+\.[0-9]+$`)
+	edgeRegx := regexp.MustCompile(`^edge-v[0-9]+\.[0-9]+\.[0-9]+$`)
+
+	if !(stableRegx.MatchString(c) || edgeRegx.MatchString(c)) {
+		return false, fmt.Errorf("misformatted release channel or version")
+	}
+
+	// get mesheryctl config
+	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		return false, fmt.Errorf("error processing config %+v", err)
+	}
+	// get current context from mesheryctl config
+	ctxCurrent, err := mctlCfg.GetCurrentContext()
+	if err != nil {
+		return false, fmt.Errorf("error fetching current context %+v", err)
+	}
+
+	verCurrent := ctxCurrent.GetVersion() //store current version
+
+	incomingVersion := strings.SplitN(c, "-", 2)[1] //get incoming version
+
+	ctxCurrent.SetVersion(incomingVersion) // temporarily set incoming version as current
+	//validate incoming version
+	err = ctxCurrent.ValidateVersion()
+	//return false if its invalid and true if it is valid and either way ,set back context version to original
+	if err != nil {
+		ctxCurrent.SetVersion(verCurrent)
+		return false, err
+	}
+	ctxCurrent.SetVersion(verCurrent)
+	return true, nil
+}
+
+//func to validate CLI argument
+func checkChannelArg(n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) != n {
+			return fmt.Errorf("release channel is a required argument in command, accepts %d arg(s), received %d", n, len(args))
+		}
+		if IsBetaOrStable(args[0]) {
+			return nil
+		} else if ok, err := validVersionCheck(args[0]); !ok {
+			return err
```

Comment:
@piyushsingariya is there supposed to be a custom error message here or are we leaving that to the person calling the function?

Reply:
Error should be returned by the function `validVersionCheck` explaining that the version is invalid.

Reply:
Okay, thanks for the clarification.

Reply:
@navendu-pottekkat , thanks for checking in, just been busy lately, but my pr almost ready as per @piyushsingariya guide lines.

Reply:
The changes asked haven't been addressed yet!